### ISSUE TITLE
Fully qualified references do not work

### ISSUE THREAD

**Issue by oberstet** [cag/sphinxcontrib-soliditydomain#3]
As this is the first issue in this repo, let me first say: kudos for creating this project! It is very useful, thank you for sharing your work. It would be awesome to have it around and further progressing ..

Now I have a little itch .. yeah, people are always asking for more once you give them sth;)

I am trying to get references to work, e.g. `:sol:func:`XBRNetwork.register` should create a link to the respective documentation that was generated from `.. autosolcontract:: XBRNetwork`, however I get:

```
overview.rst:13: WARNING: sol:func reference target not found: XBRNetwork.register
```

**Comment by cag** [comment]
I agree. It's hard to juggle everything qq

Also, thanks!

**Comment by cag** [comment]
@oberstet Should work in 0.5.0

**Comment by oberstet** [comment]
Just tested 0.5.1, and referencing contracts works as expected. cool, thanks for coming back to this!

Referencing functions has a glitch though (I think) .. using this as a test:

```
**TEST 1**: :sol:func:`register`

**TEST 2**: :sol:func:`XBRNetwork.register`

**TEST 3**: ``XBRNetwork``:sol:func:`register`
```

the first one correctly references the intended target function - there is only one named `register` in _all of the contracts_ documented. 

What if a different function with the same name exists in more than one contract?

The 2nd version does not work, but isn't that the format one usually is expected to use (fully qualified function names)?

The 3rd form is a workaround to at least render the text "fully qualified function name" - well;)

**Comment by cag** [comment]
@oberstet Yeah... I kinda just threw that together in the most basic fashion, so it is definitely not as full featured as some other domains right now :sweat_smile:
